woocommerce/woocommerce-gateway-stripe
# Add webhook for refund update action in case refund fails

closes #1997

We do not have any code to process the charge.refund.updated action. Sometimes  refunds fail async so we need to add an order note and update the order.

I was a little uncertain which status the order should have after a failed refund. I decided to go with how WC core works if you delete a refund from the edit order page. The webhook works almost exactly as [the ajax delete refund handler](https://github.com/woocommerce/woocommerce/blob/trunk/includes/class-wc-ajax.php#L2026) with a order note added and stripe fees updated. So the order will keep the refunded status.

# Testing instructions
see the ["Refund failure" critical flow](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows#refund-failure) which is basically. Checkout with CC 4000000000005126. Refund order in stripe dashboard or WC admin. Order should have a note indicating the refund failed. 

-------------------
